x-pack/functionbeat/config/config.go

```diff
@@ -134,3 +138,40 @@ func (p *ProviderConfig) Validate() error {
 	}
 	return nil
 }
+
+// MemSizeFactor64 implements a human understandable format for bytes but also make sure that all
+// values used are a factory of 64.
+type MemSizeFactor64 int
+
+// Unpack converts a size defined from a human readable format into bytes and ensure that the value
+// is a factoru of 64.
+func (m *MemSizeFactor64) Unpack(v string) error {
+	sz, err := humanize.ParseBytes(v)
+	if isRawBytes(v) {
+		cfgwarn.Deprecate("7.0.0", "size now requires a unit (KiB, MiB, etc...), current value: %s.", v)
+	}
+	if err != nil {
+		return err
+	}
+
+	if sz%64 != 0 {
+		return fmt.Errorf("number is not a factor of 64, %d bytes (user value: %s)", sz, v)
+	}
+
+	*m = MemSizeFactor64(sz)
+	return nil
+}
+
+// Megabytes return the value in megatebytes.
+func (m *MemSizeFactor64) Megabytes() int {
+	return int(*m) / 1024 / 1024
```

Comment:
why multiply by `1^20`? Doesn't it interfere with parsing if users already have configred `64MiB` ?

Reply:
> why divide by 1^20? Doesn't it interfere with parsing if users already have configred 64MiB ?

I assume you meant 2^20; otherwise, I am not sure I understand the question.

The function `Megabytes` should be renamed to `Mebibytes`. :) In the documentation of both AWS and GCP, it is stated that they require the value to be configured in MB. So I would say if someone sets `MiB` that's a config error. We can add a check to the option to make sure the value is in the correct range with the right unit. WDYT?

> Can we accept values less 64MB?

According to the pricing page of GCP, users can only configure 128MB, 256MB, 521MB, 1024MB or 2048MB. (Ref: https://cloud.google.com/functions/pricing)
In case of AWS, the following limits apply: "128 MB to 3,008 MB, in 64 MB increments". (Ref: https://docs.aws.amazon.com/lambda/latest/dg/limits.html)

There is no need to support value lower than 64MB ATM. To help the user, we can add checks to make sure the configured value is at least 128MB.

Reply:
> I assume you meant 2^20; otherwise, I am not sure I understand the question.

yeah :)

Looks like AWS and GCP parameters are somewhat different. Better handle the setting separately per provider then.

Given that the limits are under the control of the cloud providers and can potentially be changed anytime, I wonder if we should even validate the setting. When would we get an error from the providers if you use some invalid value? What state would be the setup be in in this case?

Reply:
Error is returned during setup.